apps/src/aiDifferentiation/AiDiffContainer.tsx

```diff
@@ -95,14 +112,32 @@ const AiDiffContainer: React.FC<AiDiffContainerProps> = ({
       });
   };
 
-  const onPromptSelect = (prompt: string) => {
-    const newAiMessage = {
-      role: Role.ASSISTANT,
-      chatMessageText: `You selected "${prompt}". This is a placeholder response.`,
-      status: Status.OK,
-    };
+  const onPromptSelect = (prompt: ChatPrompt) => {
+    setIsWaitingForResponse(true);
 
-    setMessageHistory([...messageHistory, newAiMessage]);
+    const body = JSON.stringify({
+      inputText: prompt.prompt,
+      lessonId: lessonId,
+      unitDisplayName: unitDisplayName,
+      sessionId: sessionId,
+    });
+    HttpClient.post(`${aiDiffChatMessageEndpoint}`, body, true, {
+      'Content-Type': 'application/json',
+    })
+      .then(response => response.json())
+      .then(json => {
+        const newAiMessage = {
+          role: Role.ASSISTANT,
+          chatMessageText: json.chat_message_text,
+          status: json.status,
+        };
+        setSessionId(json.session_id);
+        setMessageHistory(prevMessages => [...prevMessages, newAiMessage]);
+      })
+      .catch(error => console.log(error))
+      .finally(() => {
+        setIsWaitingForResponse(false);
+      });
```

Comment:
This bit looks identical to the code used above for other text messages. Let's split that out into a shared function.

Reply:
done